controllers/tenant/rhoamtenant_controller.go

```diff
@@ -0,0 +1,96 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"github.com/integr8ly/integreatly-operator/apis/v1alpha1"
+	l "github.com/integr8ly/integreatly-operator/pkg/resources/logger"
+	usersv1 "github.com/openshift/api/user/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+	"time"
+)
+
+var log = l.NewLoggerWithContext(l.Fields{l.ControllerLogContext: "tenant_controller"})
+
+// +kubebuilder:rbac:groups=integreatly.org,resources=rhoamtenant,verbs=get;list;watch
+// +kubebuilder:rbac:groups=user.openshift.io,resources=users,verbs=watch;get;list;update
+
+func New(mgr manager.Manager) (*TenantReconciler, error) {
+	restConfig := controllerruntime.GetConfigOrDie()
+	restConfig.Timeout = time.Second * 10
+
+	client, err := k8sclient.New(restConfig, k8sclient.Options{
+		Scheme: mgr.GetScheme(),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &TenantReconciler{
+		Client: client,
+		Scheme: mgr.GetScheme(),
+		mgr:    mgr,
+		log:    l.Logger{},
+	}, nil
+}
+
+type TenantReconciler struct {
+	k8sclient.Client
+	Scheme *runtime.Scheme
+	mgr    manager.Manager
+	log    l.Logger
+}
+
+func (r *TenantReconciler) Reconcile(request ctrl.Request) (ctrl.Result, error) {
+	_ = context.Background()
+	err := r.addAnnotationToUser(request.Name)
+	if err != nil {
+		r.log.Error("error in addAnnotationToUser: %s ", err)
+	}
+	return ctrl.Result{}, nil
+}
+
+func (r *TenantReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&v1alpha1.RhoamTenant{}).
+		Watches(&source.Kind{Type: &v1alpha1.RhoamTenant{}}, &handler.EnqueueRequestForObject{}).
+		Complete(r)
+}
+
+func (r *TenantReconciler) addAnnotationToUser(crName string) error {
+	crList := &v1alpha1.RhoamTenantList{}
+	err := r.Client.List(context.TODO(), crList)
+	if err != nil {
+		return fmt.Errorf("error getting RhoamTenant crs list, error: %v", err)
+	}
+	usersList := &usersv1.UserList{}
+	err = r.Client.List(context.TODO(), usersList)
+	if err != nil {
+		return fmt.Errorf("failed at finding users list, error: %v", err)
+	}
+	for _, cr := range crList.Items {
+		if cr.Name == crName {
+			for _, user := range usersList.Items {
+				if user.Name == cr.Spec.Username {
+					_, err = controllerutil.CreateOrUpdate(context.TODO(), r.Client, &user, func() error {
+						if user.Annotations == nil {
+							user.Annotations = map[string]string{}
+						}
+						user.Annotations["tenant"] = "yes"
+						return nil
+					})
+					return nil
+				}
+			}
+		}
+	}
+	log.Warning("User defined in CR was not found. Please check username in rhoamtenant CR: " + crName)
```

Comment:
Maybe return this as an error and add it to the CR status field

Reply:
Done